### PR TITLE
Remove grid gutter overrides, don't use grid for form layout

### PR DESCRIPTION
**Pull request by jrubenoff**
Refactors the grid to use percentage-based gutters for proportionally consistent styles.

To enforce consistency even further, I removed all of our gutter overrides, and modified the styles everywhere we use them. When you start using the grid within the page to lay out components, it's very difficult to maintain consistent spacing.

If we really want to keep them in, we can put the gutter overrides back, but it'll triple the size of the stylesheet since we're using absolute margins instead of padding within the container.

### CODE REVIEW

File: lib/dvl/simple_form_config.rb [hunk 1]
```diff
@@ -4,32 +4,32 @@
 }
 
 SimpleForm.setup do |config|
-  config.wrappers :horizontal, tag: 'div', class: 'grid gutter_none', error_class: 'error' do |b|
+  config.wrappers :horizontal, tag: 'div', class: 'form_horiz', error_class: 'error' do |b|
```
Comment (dobthound) [inline lib/dvl/simple_form_config.rb]: Line is too long. [91/80]

File: lib/dvl/simple_form_config.rb [hunk 2]
```diff
@@ -4,32 +4,32 @@
...
-    b.wrapper tag: 'div', class: 'item lap_nine_columns' do |ba|
+    b.wrapper tag: 'div', class: 'form_horiz_input' do |ba|
       ba.use :input
       ba.use :error, wrap_with: { tag: 'span', class: 'error_message' }
       ba.use :hint,  wrap_with: { tag: 'div', class: 'help-block' }
     end
   end
 
-  config.wrappers :vertical, tag: 'div', class: 'grid gutter_none', error_class: 'error' do |b|
+  config.wrappers :vertical, tag: 'div', class: 'form_vert', error_class: 'error' do |b|
```
Comment (dobthound) [inline lib/dvl/simple_form_config.rb]: Line is too long. [88/80]

File: vendor/assets/stylesheets/dvl/core/grid.scss [hunk 3]
```diff
@@ -1,95 +1,51 @@
-/* ================================================================== *\
-   Grids ($grids)
-\* ================================================================== */
+// Grid system
 
-$columnWidth: 8.333%;
-$gutterWidth: 2rem;
+$columnWidth: 6.25%;
+$gutterWidth: 2.0833333%;
```
Comment (ajb) [inline vendor/assets/stylesheets/dvl/core/grid.scss]: Can you use fractions so these numbers don't look like "magic numbers"? At the very least, add a comment or two

File: lib/dvl/simple_form_config.rb (hunk 1 above)
Comment (ajb) [inline lib/dvl/simple_form_config.rb]: `form_horiz` isn't really right, it's actually a "form item". Ideas?

File: vendor/assets/stylesheets/dvl/core/includes.scss [hunk 4]
```diff
@@ -48,6 +48,10 @@ $maxFontSize: 87.5% !default;
 $inputHeight: 2.286rem !default; // 32px @ 14px root
 $inputPadding: 0.3rem !default;
 
+// Viewport widths
+$lapWidth: 640px;
+$deskWidth: 960px;
```
Comment (ajb) [inline vendor/assets/stylesheets/dvl/core/includes.scss]: :+1:

File: vendor/assets/stylesheets/dvl/core/typography.scss [hunk 5]
```diff
@@ -9,7 +9,7 @@
 html { font-size: 75%; } // 12px
 @media only screen and (min-width: 320px) { html { font-size: $maxFontSize * 0.8125; } } // 13px
 @media only screen and (min-width: 480px) { html { font-size: $maxFontSize * 0.875; } } // 14px
-@media only screen and (min-width: 640px) { html { font-size: $maxFontSize * 1; } } // 16px
+@media only screen and (min-width: $lapWidth) { html { font-size: $maxFontSize * 1; } } // 16px
```
Comment (ajb) [inline vendor/assets/stylesheets/dvl/core/typography.scss]: Not really sure if this counts as `$lapWidth`, I think it makes more sense to leave it as a constant.

**Comment by ajb** [comment]
Thanks for doing this! 

A couple of todos before merging:

- [ ] It looks like nested grids aren't working? http://take.ms/XKNQP
- [ ] Can you add an example of overriding the gutters? http://take.ms/jQxID I'm fine to keep the gutter overrides out, just need an example of how to best override them on a per-component basis. (You can add the overrides to preview.scss)

File: vendor/assets/stylesheets/dvl/core/forms.scss [hunk 6]
```diff
@@ -331,10 +343,31 @@ form {
   margin-top: 2rem;
 }
 
-@media only screen and (min-width: 640px) {
-  form .horizontal_label_item {
+.form_horiz_two_inputs {
```
Comment (ajb) [inline vendor/assets/stylesheets/dvl/core/forms.scss]: Shouldn't this be a nested grid?

**Comment by jrubenoff** [comment]
So, nested grids no longer exist. Instead, you just set widths manually on each component. This can actually save a ton of markup... see this diff ([one](https://github.com/dobtco/screendoor-v2/pull/1582/files#diff-b7fde0cf0494caaec8b2a18b81d9124cL75), [two](https://github.com/dobtco/screendoor-v2/pull/1582/files#diff-2a3b4c320314b64ae65b61f9f8c703e9L1)) for an example.

At first this might sound annoying, but it's not like the old convention was more consistent... `.four_columns` in a nested grid isn't the same as `.four_columns` for the entire page. What you're really saying is, "I want all these components to be X width."

Same reasoning goes for gutter overrides. You would just manually set a `margin-right` on each component, based on what looks best. In respondent view, I've been specifying component spacing with the `$rhythm` and `$fontSize` variables for greater consistency. I plan to add similar variables to dvl-core as we redesign it.

File: lib/dvl/simple_form_config.rb [hunk 7]
```diff
@@ -4,32 +4,32 @@
 }
 
 SimpleForm.setup do |config|
-  config.wrappers :horizontal, tag: 'div', class: 'grid gutter_none', error_class: 'error' do |b|
+  config.wrappers :horizontal, tag: 'div', class: 'form_item form_item_horiz', error_class: 'error' do |b|
```
Comment (dobthound) [inline lib/dvl/simple_form_config.rb]: Line is too long. [106/80]

File: lib/dvl/simple_form_config.rb [hunk 8]
```diff
@@ -4,32 +4,32 @@
...
-    b.wrapper tag: 'div', class: 'item lap_nine_columns' do |ba|
+    b.wrapper tag: 'div', class: 'form_item_horiz_input' do |ba|
       ba.use :input
       ba.use :error, wrap_with: { tag: 'span', class: 'error_message' }
       ba.use :hint,  wrap_with: { tag: 'div', class: 'help-block' }
     end
   end
 
-  config.wrappers :vertical, tag: 'div', class: 'grid gutter_none', error_class: 'error' do |b|
+  config.wrappers :vertical, tag: 'div', class: 'form_item form_item_vert', error_class: 'error' do |b|
```
Comment (dobthound) [inline lib/dvl/simple_form_config.rb]: Line is too long. [103/80]

**Comment by ajb** [comment]
:+1: on all of that, saving markup and adding consistency are good goals.

the stuff with the margin in https://github.com/dobtco/screendoor-v2/pull/1582/files#r26613908 is going to be a deal-breaker, though, since it actively detracts from these goals by making us add un-semantic classes and wrapper divs in a lot more places.

thoughts?

**Comment by jrubenoff** [comment]
Made some changes, please review.

**Comment by ajb** [comment]
sweet! will review first thing tomorrow.

On Tuesday, March 17, 2015, Josh Rubenoff <notifications@github.com> wrote:

> Made some changes, please review.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dobtco/dvl-core/pull/59#issuecomment-82662455>.
>


-- 
Adam Becker
(951) 9-BECKER
@AdamJacobBecker

File: dist/core.css [hunk 9]
```diff
@@ -1731,7 +1731,7 @@ li.dropdown_more a {
 
   .lap_twelve_columns {
     width: 99.996%; } }
-@media only screen and (min-width: 960px) {
+@media only screen and (min-width: $deskWidth) {
```
Comment (ajb) [inline dist/core.css]: Note the filename, needs to be reverted.

**Comment by ajb** [comment]
Before I review, can you re-state the desired outcome of this PR? I know that we're not gonna create something 100% perfect in every way, but what are we trying to accomplish with this PR?

File: lib/dvl/simple_form_config.rb (hunk 7 above)
Comment (dobthound) [inline lib/dvl/simple_form_config.rb]: Line is too long. [106/80]

**Comment by ajb** [comment]
Re-added variables and removed gutter overrides. Merging!

**Pull request merged**